src/components/data_submission/ds_common.js

```diff
@@ -0,0 +1,341 @@
+          val,
+          span({ className: 'glyphicon glyphicon-remove', style: { marginLeft: '8px' } })
+        ]);
+      })
+    )
+  ]);
+};
+
+const formInputSelect = (config) => {
+  const {
+    id, title, disabled, errors,
+    selectOptions, searchPlaceholder, ariaDescribedby
+  } = config;
+
+  return h(SearchSelectOrText, {
+    id,
+    'aria-describedby': ariaDescribedby,
+    onPresetSelection: async (selection) => onFormInputChange(config, selection),
+    onManualSelection: (selection) => onFormInputChange(config, selection),
+    options: selectOptions.map((x) => { return { key: x, displayText: x }; }),
+    searchPlaceholder: searchPlaceholder || `Search for ${title}...`,
+    className: 'form-control',
+    disabled, errored: errors[id]
+  });
+};
+
+const formInputCheckbox = (config) => {
+  const {
+    id, disabled, errors, toggleText,
+    formInfo, defaultValue, ariaDescribedby
+  } = config;
+
+  return div({ className: 'checkbox' }, [
+    input({
+      type: 'checkbox',
+      id: `cb_${id}_${toggleText}`,
+      checked: get(id, formInfo) === undefined ? defaultValue : get(id, formInfo),
+      className: 'checkbox-inline',
+      'aria-describedby': ariaDescribedby,
+      onChange: (event) => onFormInputChange(config, event.target.checked),
+      disabled
+    }),
+    label({
+      className: `regular-checkbox ${errors[id] ? 'errored' : ''}`,
+      htmlFor: `cb_${id}_${toggleText}`,
+      style: {
+        fontWeight: 'normal',
+        fontStyle: 'italic'
+      }
+    }, [toggleText])
+  ]);
+};
+
+const formInputSlider = (config) => {
+  const {
+    id, disabled, toggleText, defaultValue,
+    formInfo
+  } = config;
+
+  return div({ style: { ...styles.flexRow, justifyContent: 'unset', alignItems: 'center' } }, [
+    label({ className: 'switch', htmlFor: `cb_${id}_${toggleText}` }, [
+      input({
+        type: 'checkbox',
+        id: `cb_${id}_${toggleText}`,
+        checked: get(id, formInfo) === undefined ? defaultValue : get(id, formInfo),
+        className: 'checkbox-inline',
+        onChange: (event) => onFormInputChange(config, event.target.checked),
+        disabled
+      }),
+      div({className: 'slider round'}),
+    ]),
+    div({
+      style: {
+        marginLeft: 15,
+        fontStyle: 'italic'
+      }
+    }, [toggleText])
+  ]);
+};
+
+//---------------------------------------------
+// Main Components
+//---------------------------------------------
+/*
+* Config options:
+* id, title, description
+* type (ENUM: 'text', 'multitext', 'select', 'sliding-checkbox')
+*  * type == select
+*    * selectOptions: [{key: string, displayText: string}]
+*    * searchPlaceholder
+*  * type == 'checkbox'
+*    * toggleText
+*    * checkboxType
+* required, disabled
+* placeholder, defaultValue,
+* style (for the formControl component)
+* inputStyle (for the input element)
+* onChange,
+* errors, setErrors,
+* formInfo, setFormInfo
+*/
+export const FormField = (config) => {
```

Comment:
I have so many thoughts about this! We should try to do as much as possible through css for normalizing the appearances of form fields, but where there is a necessary html component (like checkboxes and sliders), i would definitely want to create generic form components for it!

I think the components folder is really hard to browse for existing items though, and could use a little restructuring? Like, generic reuseable components should be grouped, and specific (potentially single purpose use) components shoudl also be grouped separately.

Maybe we could make a common/ colder inside components? Or it could exist outside of components as it's own `common/` too?